### PR TITLE
Sesame Search Provider

### PR DESCRIPTION
**Pull request by yasanglass**
This PR currently just contains the removed commit from #2672 that added Sesame search provider (on top of the commits in #2672). I will adjust the code based on the discussion made on #2672 and this draft PR is only created for further discussion around the topic while it's being worked on.

### CODE REVIEW

**Comment by yasanglass** [comment]
Ready for review! As discussed with @patrykmichalik, the core functionality is ready & it just requires some UI/UX changes which will be done by the core Lawnchair team instead of me. 

Thanks, @patrykmichalik!

File: lawnchair/src/app/lawnchair/qsb/providers/QsbSearchProvider.kt [hunk 1]
```diff
@@ -65,6 +69,51 @@ open class QsbSearchProvider(
...
+     * Should only be used for [QsbSearchProviderType]s that are downloadable.
+     */
+    fun isDownloaded(context: Context): Boolean {
+        return try {
+            context.packageManager.getPackageInfo(packageName, PackageManager.GET_META_DATA)
+            true
+        } catch (e: PackageManager.NameNotFoundException) {
+            false
+        }
+    }
```
Comment (ghost) [inline lawnchair/src/app/lawnchair/qsb/providers/QsbSearchProvider.kt]: I would suggest using the `PackageManager#isPackageInstalled` extension function here. See lines 137 and 138 of `IconPackPreferences.kt` for an example.

**Review by ghost — CHANGES_REQUESTED** [review]
Hello! Here’s one minor suggestion.

File: lawnchair/src/app/lawnchair/qsb/providers/QsbSearchProvider.kt (hunk 1 above)
Comment (yasanglass) [reply lawnchair/src/app/lawnchair/qsb/providers/QsbSearchProvider.kt]: Done!

**Comment by ghost** [comment]
Thanks, @yasandev and @DariaHamrahPaytakht! As we’ve discussed, I’ll be merging the changes from this pull request into a feature branch.

**Pull request merged**